lib/widgets/components/receipt/receipts_list_view.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:coffeecard/base/style/colors.dart';
+import 'package:coffeecard/blocs/receipt/receipt_cubit.dart';
+import 'package:coffeecard/models/receipts/receipt.dart';
+import 'package:coffeecard/widgets/components/receipt/receipt_list_entry.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class ReceiptsListView extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Expanded(
+      child: Container(
+        color: AppColor.white,
+        child: BlocBuilder<ReceiptCubit, ReceiptState>(
+          builder: (context, state) {
+            if (state.status.isInitial) {
```

Comment:
Consider if it could be refactored to minimize all those ifs

Reply:
Don't see a way to remove the first one either. The `if else` could be removed by changing the `ReceiptListEntry` to be more generic and receive the transaction type, since it already has a check inside itself whether it is a purchase. Whether that would be preferable is a different matter.